RevenueCatUI/PaywallView.swift

```diff
@@ -100,10 +104,13 @@ public struct PaywallView: View {
         fonts: PaywallFontProvider = DefaultPaywallFontProvider(),
         displayCloseButton: Bool = false,
         introEligibility: TrialOrIntroEligibilityChecker?,
-        purchaseHandler: PurchaseHandler?
+        purchaseHandler: PurchaseHandler?,
+        refreshSubscriptions: @escaping () async throws -> Void
     ) {
         self._introEligibility = .init(wrappedValue: introEligibility ?? .default())
-        self._purchaseHandler = .init(wrappedValue: purchaseHandler ?? .default())
+        let installedPurchaseHandler = purchaseHandler ?? .default()
+        installedPurchaseHandler.refreshSubscriptions = refreshSubscriptions
+        self._purchaseHandler = .init(wrappedValue: installedPurchaseHandler)
```

Comment:
Thinking on the scenario where you have a valid `PurchaseHandler` passed with its `refreshSubscriptions`.
Shouldn't, in that case, use that `refreshSubscriptions` instead of the one passed in the function?

Reply:
but in that case this argument would not do anything?

init of purchaseHandler inside View is not great, yet we have to deal with it 🤷

Reply:
> init of purchaseHandler inside View is not great, yet we have to deal with it 🤷

Yes I understood

> but in that case this argument would not do anything?

Eh, good question.
Having this `purchaseHandler` in the view init will always lead you to ignore one of the eventual 2 `purchaseHandler` you are passing (one explicit and one inside `PurchaseHandler`).
So, imo I would give precedence to the `PurchaseHandler` argument (if any) that has all the info inside it, and treat `purchaseHandler` argument as a fallback in this case.

Reply:
I've changed 'PurchaseHandler' to be non-optional now.